sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -1920,24 +1929,70 @@ mod tests {
         assert!(comp_res.warnings.is_empty() && comp_res.errors.is_empty());
     }
 }
-fn disallow_opcode(op: &Ident) -> CompileResult<()> {
-    let mut errors = vec![];
 
-    match op.as_str().to_lowercase().as_str() {
-        "ji" => {
-            errors.push(CompileError::DisallowedJi { span: op.span() });
-        }
-        "jnei" => {
-            errors.push(CompileError::DisallowedJnei { span: op.span() });
-        }
-        "jnzi" => {
```

Comment:
One thing I'm concerned about is the UX of this change. Consider a user who is reading the opcode specs. They think they can type any opcode in an asm block. Currently, if they type a `ji`, the compiler will inform them that specifically that opcode is disallowed. With this change, the error will be `unrecognized opcode`. That error message is confusing, as it certainly is an opcode.

I think we should either parse disallowed opcodes and report a nicer error about them, or have extensive documentation on allowed and disallowed opcodes in the Sway Book. Preferably the former as it maintains the current UX.

Reply:
That's a good point and you're absolutely right. This list is also incomplete at the moment and that code wasn't triggering anyways because the parser won't allow you to have a `ji` for example. So, we're in no worse place than before.

What I'm going to do is address this UX issue in a separate PR where I allow all opcodes in the parser but disallow them in type checking. Let me open an issue about that and I'll make a follow up PR.

Reply:
Sounds good, I'll approve this for now. You're right that this is actually a no-op currently, I missed that.